### ISSUE TITLE
[feat] Support mongodb replica set for mongo shared clusters support

### ISSUE THREAD

**Issue by robertsLando** [mcollina/mqemitter-mongodb#19]
Consider this question: https://developer.mongodb.com/community/forums/t/mongodb-pub-sub-using-tailable-cursors-in-nodejs-with-clusters/3027/4

This package could have some problems when using mongodb running on shared clusters. 

In the question linked there are all the tips to use to make the support, maybe a new mqemitter packege could be created for that instead of making this supporting both

**Comment by mcollina** [comment]
Is this fixed?

**Comment by robertsLando** [comment]
Nope, this is an idea for a future development based on the question I linked in previous comment. 

Tailable cursors may not be the best solution with MONGODB SHARED CLUSTERS

My PR has fixed mqemitter when using with NODEJS CLUSTERS.

---

Daniel - Software Engineer

Support me at:
    Github sponsors

**Comment by mcollina** [comment]
I don't know if it's possible to use something else on top of mongo. Go ahead and write another package if it would be a total rewrite of this one.

**Comment by robertsLando** [comment]
@mcollina Yes, sure :+1:  Keep this open in the meanwhile

**Comment by 1977744311** [comment]
@mcollina  @robertsLando  I tried to start two node services to connect to the same mongo client. I found that the latest task will be consumed twice. Is this normal? How can I solve it?

**Comment by robertsLando** [comment]
Code snippet to reproduce the issue?

**Comment by 1977744311** [comment]
**You can try to pull the code to execute locally**
`git clone git@github.com:1977744311/issue-mq.git`
**And you can do:**
`npm install`
`npm run issue`
Then you can see two services in pm2, You can visit the address of one of the services http://127.0.0.1:3000/api/emit, you can find the result of two emits in `pm2 logs`
How should I avoid this, or is this package not supporting different instances connecting to the same database

**Comment by robertsLando** [comment]
> I found that the latest task will be consumed twice. Is this normal? How can I solve it?

What do you mean with 'consumed twice' ? Do you mean that both services receive it once or that each service receive it twice?

**Comment by 1977744311** [comment]
both services receive it once

**Comment by robertsLando** [comment]
> both services receive it once

That's the expected behaviour so. mqemitter is used exactly for that purpose, share messages across multiple instances. If you don't want all instances to receive a specific message you should create your own logic

**Comment by 1977744311** [comment]
> > both services receive it once
> 
> That's the expected behaviour so. mqemitter is used exactly for that purpose, share messages across multiple instances. If you don't want all instances to receive a specific message you should create your own logic

okey，thanks for your answer
